src/include/duckdb/common/vector_operations/quaternary_executor.hpp

```diff
@@ -0,0 +1,211 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/vector_operations/quaternary_executor.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+
+#include <functional>
+
+namespace duckdb {
+
+struct QuaternaryLambdaWrapper {
+	template <class FUN, class A_TYPE, class B_TYPE, class C_TYPE, class D_TYPE, class RESULT_TYPE>
+	static inline RESULT_TYPE Operation(FUN fun, A_TYPE a, B_TYPE b, C_TYPE c, D_TYPE d, ValidityMask &mask,
+	                                    idx_t idx) {
+		return fun(a, b, c, d);
+	}
+};
+
+struct QuaternaryLambdaWrapperWithNulls {
+	template <class FUN, class A_TYPE, class B_TYPE, class C_TYPE, class D_TYPE, class RESULT_TYPE>
+	static inline RESULT_TYPE Operation(FUN fun, A_TYPE a, B_TYPE b, C_TYPE c, D_TYPE d, ValidityMask &mask,
+	                                    idx_t idx) {
+		return fun(a, b, c, d, mask, idx);
+	}
+};
+
+struct QuaternaryExecutor {
```

Comment:
The `UnaryExecutor`, `BinaryExecutor` and `TernaryExecutor` exist to generate optimized code for primitive types/different vector types to create as optimal code as possible (at the expense of an increased code footprint).

I think adding this class is a bit overkill for the `bar` function - could we instead use `GenericExecutor::ExecuteQuaternary`?

Reply:
I didn't notice it.
`TernaryExecutor` and `QuaternaryExecutor` are replaced with `GenericExecutor`.